### PR TITLE
Sonobi: add native support to sonobi adapter

### PR DESCRIPTION
**Pull request by bansawbanchee**

### CODE REVIEW

File: static/bidder-info/sonobi.yaml [hunk 1]
```diff
@@ -7,11 +7,16 @@ capabilities:
...
+      - native
   app:
     mediaTypes:
       - banner
       - video
+      - native
 userSync:
+  iframe:
+    url: "https://sync.go.sonobi.com/uc.html?gdpr={{.GDPR}}&gdpr_consent={{.GDPRConsent}}&us_privacy={{.USPrivacy}}&loc={{.RedirectURL}}"
+    userMacro: "[UID]"
```
Comment (gargcreation1992) [inline static/bidder-info/sonobi.yaml:19]: usersync works.

**Review by gargcreation1992 — DISMISSED** [review]

File: adapters/sonobi/sonobi.go [hunk 2]
```diff
@@ -140,6 +140,9 @@ func getMediaTypeForImp(impID string, imps []openrtb2.Imp) (openrtb_ext.BidType,
 			if imp.Banner == nil && imp.Video != nil {
 				mediaType = openrtb_ext.BidTypeVideo
 			}
+			if imp.Banner == nil && imp.Video == nil && imp.Native != nil {
+				mediaType = openrtb_ext.BidTypeNative
```
Comment (onkarvhanumante) [inline adapters/sonobi/sonobi.go:144]: Code path is not covered.  Should add json framework tests for `Native` media type.

Comment (gargcreation1992) [reply adapters/sonobi/sonobi.go:144]: @bansawbanchee Suggesting you to please use Mtype field set by your adapter response. Here is an example -
```
func getBidType(bid openrtb2.Bid) (openrtb_ext.BidType, error) {
	switch bid.MType {
	case openrtb2.MarkupBanner:
		return openrtb_ext.BidTypeBanner, nil
	case openrtb2.MarkupVideo:
		return openrtb_ext.BidTypeVideo, nil
	case openrtb2.MarkupNative:
		return openrtb_ext.BidTypeNative, nil
	}

	return "", &errortypes.BadInput{
		Message: fmt.Sprintf("Could not define media type for impression: %s", bid.ImpID),
	}
}
```

**Comment by bansawbanchee** [comment]
Went ahead and closed this on and combined it into our newest pull request: https://github.com/prebid/prebid-server/pull/3889

**Pull request closed without merging**